toolchain/lex/tokenized_buffer_benchmark.cpp

```diff
@@ -226,15 +226,66 @@ auto GetSymbolTokenTable() -> llvm::ArrayRef<TokenKind> {
   return symbol_token_table_storage;
 }
 
-// Compute a random sequence of mixed symbols, keywords, and identifiers, with
-// percentages of each according to the parameters.
-auto RandomMixedSeq(int symbol_percent, int keyword_percent) -> std::string {
-  CARBON_CHECK(0 <= symbol_percent && symbol_percent <= 100)
-      << "Must be a percent: [0, 100].";
-  CARBON_CHECK(0 <= keyword_percent && keyword_percent <= 100)
-      << "Must be a percent: [0, 100].";
-  CARBON_CHECK((symbol_percent + keyword_percent) <= 100)
-      << "Cannot have >100%.";
+struct RandomSourceOptions {
+  int symbol_percent = 0;
+  int keyword_percent = 0;
+  int numeric_literal_percent = 0;
+  int string_literal_percent = 0;
+
+  int tokens_per_line = NumTokens;
+
+  int comment_line_percent = 0;
+  int blank_line_percent = 0;
+
+  void Validate() {
+    CARBON_CHECK(0 <= symbol_percent && symbol_percent <= 100);
+    CARBON_CHECK(0 <= keyword_percent && keyword_percent <= 100);
+    CARBON_CHECK(0 <= numeric_literal_percent &&
+                 numeric_literal_percent <= 100);
+    CARBON_CHECK(0 <= string_literal_percent && string_literal_percent <= 100);
+    CARBON_CHECK((symbol_percent + keyword_percent + numeric_literal_percent +
+                  string_literal_percent) <= 100);
+
+    CARBON_CHECK(tokens_per_line <= NumTokens);
+    CARBON_CHECK(NumTokens % tokens_per_line == 0)
+        << "Tokens per line of " << tokens_per_line
+        << " does not divide the number of tokens " << NumTokens;
+
+    CARBON_CHECK(0 <= comment_line_percent && comment_line_percent <= 100);
+    CARBON_CHECK(0 <= blank_line_percent && blank_line_percent <= 100);
+    CARBON_CHECK((comment_line_percent + blank_line_percent) < 100);
```

Comment:
I wonder if a helper function would help:
```suggestion
    auto is_percentage = [](int n) { return 0 <= n && n <= 100; };
    CARBON_CHECK(is_percentage(symbol_percent));
    CARBON_CHECK(is_percentage(keyword_percent));
    CARBON_CHECK(is_percentage(numeric_literal_percent));
    CARBON_CHECK(is_pecentage(string_literal_percent));
    CARBON_CHECK(is_percentage(symbol_percent + keyword_percent +
                               numeric_literal_percent + string_literal_percent));

    CARBON_CHECK(tokens_per_line <= NumTokens);
    CARBON_CHECK(NumTokens % tokens_per_line == 0)
        << "Tokens per line of " << tokens_per_line
        << " does not divide the number of tokens " << NumTokens;

    CARBON_CHECK(is_percentage(comment_line_percent));
    CARBON_CHECK(is_percentage(blank_line_percent));
    CARBON_CHECK(comment_line_percent + blank_line_percent < 100);
```
A comment on the last line explaining why the sum must be < 100% would have helped me here. (I see below that the reason is we need some lines to put the tokens on!)